### DISCUSSION TITLE
how to set a anisotropic eigenstrain

### DISCUSSION THREAD

**Discussion by wangzhaohao** [idaholab/moose#23595] [Q&A General] [answered]
Hi, everyone. I need to set a anisotropic eigenstrain (RZ). I find in `thermal expansion` there used `GenericMaterialProperty<is_ad> _eigenstrain`  and use `addIa` to set the isotropic eigenstrain.  but, sorry i don't find another  member function to set in anisotropic eigenstrain. Thanks for you advice.
zhaohao wang

**Comment by hugary1995** [comment, marked as answer]
There's nothing preventing your from setting the components of a RankTwoTensor. Please refer to the doxygen.
